README.md

```diff
@@ -1,289 +1,610 @@
 
-## Terminology definition
+  <dt>Application</dt>
+  <dd>Any process, running within a container.  Examples include a Spring Boot application, a NodeJS Express application, or a Ruby Rails application.  <b>Note:</b> This is different than an umbrella application as defined by the Kubernetes SIG, which refers to a set of micro-services.</dd>
+
+  <dt>Service Binding</dt>
+  <dd>The act of or representation of the action of providing information about a Service to an Application</dd>
+
+  <dt>ConfigMap</dt>
+  <dd>A Kubernetes <a href="https://kubernetes.io/docs/concepts/configuration/secret/">ConfigMap</a></dd>
+
+  <dt>Secret</dt>
+  <dd>A Kubernetes <a href="https://kubernetes.io/docs/concepts/configuration/configmap/">Secret</a></dd>
+</dl>
+
+# Provisioned Service
+
+A Provisioned Service resource **MUST** define a `.status.binding.name` which is a `LocalObjectReference` to a `Secret`.  The `Secret` **MUST** be in the same namespace as the resource.  The `Secret` **SHOULD** contain a `type` entry with a value that identifies the abstract classification of the binding.  It is **RECOMMENDED** that the `Secret` also contain a `provider` entry with a value that identifies the provider of the binding.  The `Secret` **MAY** contain any other entry.
+
+## Resource Type Schema
+
+```yaml
+status:
+  binding:
+    name:  # string
+```
+
+## Example Resource
+
+```yaml
+...
+status:
+  ...
+  binding:
+    name: production-db-secret
+```
+
+## Well-known Secret Entries
+
+Other than the recommended `type` and `provider` entries, there are no other reserved `Secret` entries.  In the interests of consistency, if a `Secret` includes any of the following entry names, the entry value **MUST** meet the specified requirements:
+
+| Name | Requirements
+| ---- | ------------
+| `host` | A DNS-resolvable host name or IP address
+| `port` | A valid port number
+| `uri` | A valid URI as defined by [RFC3986](https://tools.ietf.org/html/rfc3986)
+| `username` | A string-based username credential
+| `password` | A string-based password credential
+| `certificates` | A collection of PEM-encoded X.509 certificates, representing a certificate chain used in mTLS client authentication
+| `privateKey` | A PEM-encoded private key used in mTLS client authentication
+
+`Secret` entries that do not meet these requirements **MUST** use different entry names.
+
+## Example Secret
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: production-db
+stringData:
+  type: mysql
+  provider: bitnami
+  host: localhost
+  port: 3306
+  username: root
+  password: root
+```
+
+# Application Projection
+
+A Binding `Secret` **MUST** be volume mounted into a container at `$SERVICE_BINDINGS_ROOT/<binding-name>` with directory names matching the name of the binding.  Binding names **MUST** match `[a-z0-9\-\.]{1,253}`.  The `$SERVICE_BINDINGS_ROOT` environment variable **MUST** be declared and can point to any valid file system location.
```

Comment:
Is `binding-name` the secret name?  If the secret is the output of some provisioning system, I could see this being a challenging UX (e.g. could be a uuid).

Reply:
Later in the spec it says: `If a .spec.name is set, the directory name relative to $SERVICE_BINDING_ROOT MUST be its value. If a .spec.name is not set, the directory name relative to $SERVICE_BINDING_ROOT SHOULD be the value of .metadata.name.`

Reply:
Agreed that generate secret names are problematic.  The plan is to have a sensible default with a user-override where appropriate.  Also, in practice, utilizing the `name` of service binding, rather than its `type` is incredibly brittle.  It makes any logic related to the service name, more or less non-portable.  Thought out Heroku's and Cloud Foundry's history, we always recommended, and see near universal adoption, of consuming services by type rather than name.  I believe that this spec definition here enables and encourages this behavior (hence the requirement for an application projection to have a `type`), and gives users the ability to rename if they choose to go that direction.

Reply:
The binding name is orthogonal to the secret name. It is only used as the directory under $SERVICE_BINDINGS_ROOT to mount the secret, and must be unique within the container. The secret name and binding name are intentionally decoupled.

@arthurdm that phrasing is confusing. The way I think of it:
1. `<binding-name>` is the directory under `$SERVICE_BINDING_ROOT` where the secret is mounted
2. `<binding-name>` defaults to the ServiceBinding's `.metadata.name`, but may be overridden with `.spec.name`

Reply:
Addressed via 0ceff72